components/mdc/TextInput/MoneyInput.svelte

```diff
@@ -31,7 +31,7 @@ $: isLowerThanMinValue = minValue && internalValue < minValue
 $: showErrorIcon = hasExceededMaxValue || isLowerThanMinValue || hasExceededMaxLength || valueNotDivisibleByStep
 $: error = showErrorIcon || (hasFocused && hasBlurred && required && !internalValue)
 $: showCounter = maxlength && valueLength / maxlength > 0.85
-$: valueNotDivisibleByStep = internalValue && (internalValue / Number(step)) % 1 !== 0
+$: valueNotDivisibleByStep = internalValue && (internalValue / Number(step)).toFixed(13) % 1 !== 0
```

Comment:
Number % 1 is 0 by definition. Something seems weird with this line of code.

Reply:
The problem is that js uses floats, so numbers division has low accuracy with fractions. Since step is a fraction (0.01), doing modulo doesn't work too well.

If we're only using currency, and assume steps are going to always be 1x10^n, then we could make a method to get last digit place, then ensure internalValue's last digit place is less than or equal to step's last digit place.

That falls through if steps are not always 1x10^n (e.g. 0.2).